File: app/src/main/java/com/my/televip/virtuals/ui/Cells/HeaderCell.java
package com.my.televip.virtuals.ui.Cells;

import android.content.Context;
import android.view.View;

import com.my.televip.Class.ClassLoad;
import com.my.televip.Class.ClassNames;
import com.my.televip.obfuscate.Obfuscate;

import de.robv.android.xposed.XposedHelpers;

public class HeaderCell {

    public Object headerCell;

    public HeaderCell(Context context){
        headerCell = XposedHelpers.newInstance(ClassLoad.getClass(ClassNames.HEADER_CELL), context);
    }

    public HeaderCell(Object obj){
       headerCell = obj;
    }

    public View getView(){
        return (View) headerCell;
    }

    public void setText(CharSequence text){
        XposedHelpers.callMethod(headerCell, Obfuscate.getMethodName("HeaderCell","setText"), text);
    }

}

File: app/src/main/java/com/my/televip/virtuals/ui/Cells/HeaderCell.kt
package com.my.televip.virtuals.ui.Cells

import android.content.Context
import android.view.View
import com.my.televip.Class.ClassLoad
import com.my.televip.Class.ClassNames
import com.my.televip.obfuscate.Obfuscate
import de.robv.android.xposed.XposedHelpers

class HeaderCell {

    @JvmField
    var headerCell: Any? = null

    constructor(context: Context?) {
        val hcClass = ClassLoad.getClass(ClassNames.HEADER_CELL)
        headerCell = XposedHelpers.newInstance(hcClass, context)
    }

    constructor(obj: Any?) {
        headerCell = obj
    }

    val view: View?
        get() = headerCell as? View

    fun setText(text: CharSequence?) {
        val target = headerCell ?: return
        XposedHelpers.callMethod(target, Obfuscate.getMethodName("HeaderCell", "setText"), text)
    }
}
